pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto

```diff
@@ -668,6 +668,7 @@ enum ContainerState {
     CONTAINER_RUNNING = 1;
     CONTAINER_EXITED  = 2;
     CONTAINER_UNKNOWN = 3;
+    CONTAINER_OOMKILLED = 4;
```

Comment:
I agree that kuberuntime code should not depend on the `Reason` field, which is just a human-readable string

To be fair, the code was added in https://github.com/kubernetes/kubernetes/pull/39341, after CRI was already defined. It wasn't intentional. In the code referred, if the exit code is non-zero (or OOMKilled), kuberuntime will fall back to logs. In that case, i think "non-zero exit code" itself might be enough like @Random-Liu said?